common/scala/src/main/scala/org/apache/openwhisk/common/Logging.scala

```diff
@@ -414,6 +414,28 @@ object LoggingMarkers {
       LogMarkerToken(loadbalancer + controllerInstance.asString, s"memory${actionType}Inflight", counter)(
         MeasurementUnit.none)
 
+  // Counter metrics for completion acks in load balancer
+  sealed trait completionAckType { def asString: String = ??? }
```

Comment:
I don't think you need to implement the method here? Just define it in the trait, `???` will throw an exception at runtime.

Reply:
Changed to variant without `???` but only declaring `asString`. Compiler will then complain of a sub-type does not implement `asString`.